pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,263 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package instancetype
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/create/params"
+)
+
+const (
+	Instancetype = "instancetype"
+
+	CPUFlag             = "cpu"
+	MemoryFlag          = "memory"
+	GPUFlag             = "gpu"
+	HostDeviceFlag      = "hostdevice"
+	IOThreadsPolicyFlag = "iothreadspolicy"
+	NameFlag            = "name"
+	ClusterFlag         = "cluster"
+)
+
+type instancetype struct {
+	name   string
+	cpu    int
+	memory string
+
+	gpus            []string
+	hostDevices     []string
+	ioThreadsPolicy string
+	cluster         bool
+}
+
+type GPU struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+type HostDevice struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+var flags = []string{
+	GPUFlag,
+	HostDeviceFlag,
+	IOThreadsPolicyFlag,
+}
+
+type optionFn func(*instancetype, *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error
+
+var optFns = map[string]optionFn{
+	GPUFlag:             withGPUs,
+	HostDeviceFlag:      withHostDevices,
+	IOThreadsPolicyFlag: withIOThreadsPolicy,
+}
+
+func NewInstancetypeCommand() *cobra.Command {
+	c := defaultCreateInstancetype()
+	cmd := &cobra.Command{
+		Use:     Instancetype,
+		Short:   "Create Instancetype manifest.",
+		Example: c.usage(),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return c.run(cmd)
+		},
+	}
+	cmd.Flags().StringVar(&c.name, NameFlag, c.name, "Specify the name of the Instancetype.")
+	cmd.Flags().IntVar(&c.cpu, CPUFlag, c.cpu, "Specify the count of CPUs of the Instancetype.")
+	cmd.Flags().StringVar(&c.memory, MemoryFlag, c.memory, "Specify the amount of memory of the Instancetype.")
+	cmd.Flags().StringVar(&c.ioThreadsPolicy, IOThreadsPolicyFlag, c.ioThreadsPolicy, "Specify IOThreadsPolicy to be used. Only valid values are \"auto\" and \"shared\".")
+	cmd.Flags().BoolVar(&c.cluster, ClusterFlag, false, "Specify if clusterInstancetype should be created. By default instancetype is created.")
```

Comment:
I suggested we default to the cluster scoped type as that's the default in the matchers.

Reply:
@lyarwood @opokornyy IMO I would do the opposite - default to a namespaced object, it seems more secure. Cluster-wide objects require higher permissions. But I guess its opinionated. Let see the feedback from the field

Reply:
I've always argued that the use case for the namespaced variants is a little weak, that's why the matchers and other logic default to looking up the cluster wide variants first. Let's keep the current default of clusterwide and definitely seek feedback from users in the coming releases on this, it can always change.